### PR TITLE
Add a basic gem.rbi to lib/ruby.

### PR DESCRIPTION
**Pull request by connorshea**
The goal here is to resolve https://github.com/sorbet/sorbet/issues/1455

There are a few problems with this as it is now:

- Almost everything is untyped
- Lots of `_` parameters
- This is specifically for RubyGems 3.x
- I'm not sure if `gem.rbi` should be part of the `ruby/` directory or not?
- I'm not sure if `gem.rbi` should actually be `rubygems.rbi`?
- Can Sorbet actually determine the user's RubyGems version?
- There may be some minor errors, I generated this with Sord, but I've fixed a few of its quirks manually and generally I think it's been pretty well battle-tested at this point.

Adding this RBI to my `sorbet/` directory and then regenerating `hidden.rbi` drops it down from 7419 lines to 3386 lines.

### CODE REVIEW

**Comment by connorshea** [comment]
Also, this exists already: https://github.com/sorbet/sorbet/blob/719eb639e535860a3f1e3288fed3dcd47586eb5e/rbi/stdlib/gem.rbi

So that's probably a problem

**Comment by connorshea** [comment]
Any ideas on fixing this last error? 🤔 
```
Connors-MacBook-Pro:sorbet-typed connorshea$ ruby .ci/run.rb 
lib/bundler/all/bundler.rbi:1211: Type Elem declared by parent T.class_of(Gem::Specification) must be re-declared in T.class_of(Bundler::EndpointSpecification) https://srb.help/5036
    1211 |class Bundler::EndpointSpecification < Gem::Specification
                ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
    https://github.com/sorbet/sorbet/tree/719eb639e535860a3f1e3288fed3dcd47586eb5e/rbi/stdlib/gem.rbi#L207: Elem declared in parent here
     207 |class Gem::Specification < Gem::BasicSpecification
                ^^^^^^^^^^^^^^^^^^
Errors: 1
```

**Comment by jasnow** [comment]
Did you try "srb tc--suggest-typed" or "srb tc -a" ?

**Comment by jez** [comment]
@connorshea I would propose removing the autogenerated sigs that have untyped. Putting them there locks in the sigs for everyone else, whereas by omitting them, people can one-off sig individual methods to have better types in the mean time while they're submitting a patch to add proper types upstream.

**Comment by connorshea** [comment]
Removed all the untyped stuff 👍

**Comment by elliottt** [comment]
It looks like the error with the bundler rbi is due to the bundler.rbi that exists in sorbet https://github.com/sorbet/sorbet/blob/master/rbi/gems/bundler.rbi

**Comment by connorshea** [comment]
Hmm, why do you think so?

**Comment by elliottt** [comment]
The build failure for your most recent commit says that you need to declare `Elem`, but you're already doing that as of a86b6ec2379d5cbc6b880cddc70c30f822a21529.

**Comment by connorshea** [comment]
Any ideas on how to fix it?

File: lib/ruby/all/gem.rbi [hunk 1]
```diff
@@ -0,0 +1,4568 @@
...
+    def suggestion; end
+
+    def suggestion=(suggestion); end
+  end
+
+  module Security
+    AlmostNoSecurity = T.let(nil, T.untyped)
+    DIGEST_NAME = T.let(nil, T.untyped)
+    EXTENSIONS = T.let(nil, T.untyped)
+    H
```
Comment (elliottt) [inline lib/ruby/all/gem.rbi]: Did you mean to `extend Enumerable` here? That's likely what's causing problems with the bundler rbi.

Comment (connorshea) [reply lib/ruby/all/gem.rbi]: oh wow, good catch. I'm not entirely sure how that got in there.

Comment (elliottt) [reply lib/ruby/all/gem.rbi]: You should also be able to remove the changes to the bundler rbi now.

Comment (connorshea) [reply lib/ruby/all/gem.rbi]: 👍 done

**Pull request merged**